include/seqan/blast/blast_report_out.h

```diff
@@ -0,0 +1,893 @@
+                                     TChar1 const & char1,
+                                     TChar2 const & char2,
+                                     BlastReport const & /*tag*/)
+{
+    // TODO(h4nn3s): this function could be replaced by a matrix, which would be a lot faster
+    // but it would also require a matrix for every scoringScheme
+
+    if (char1 == char2)
+        write(stream, char1);
+    else if ((char1 == '-') || (char2 == '-'))
+        write(stream, ' ');
+    else if (score(seqanScheme(context.scoringScheme), char1, char2) > 0)
+        write(stream, '+');
+    else
+        write(stream, ' ');
+}
+
+// ----------------------------------------------------------------------------
+// Function _writeAlignmentBlock
+// ----------------------------------------------------------------------------
+
+template <typename TStream,
+          typename TScore,
+          typename TConString,
+          typename TQId,
+          typename TSId,
+          typename TPos,
+          typename TAlign,
+          BlastProgram p,
+          BlastTabularSpec h>
+inline void
+_writeAlignmentBlock(TStream & stream,
+                     BlastIOContext<TScore, TConString, p, h> & context,
+                     BlastMatch<TQId, TSId, TPos, TAlign> const & m,
+                     BlastReport const & /*tag*/)
+{
+    TPos const   windowSize  = 60;
+
+    char            buffer[40]  = "";
+
+    TPos            aPos        = 0; // position in alignment
+    TPos            qPos        = 0; // position in query (without gaps)
+    TPos            sPos        = 0; // position in subject (without gaps)
+    // the latter two might become negative in below calculation, but it is safe even for unsigned types
+
+    TPos            effQStart   = m.qStart;
+    TPos            effQEnd     = m.qEnd;
+    TPos            effSStart   = m.sStart;
+    TPos            effSEnd     = m.sEnd;
+
+    _untranslateQPositions(effQStart, effQEnd, m.qFrameShift, m.qLength, context.blastProgram, BlastProgramTag<p>());
+    _untranslateSPositions(effSStart, effSEnd, m.sFrameShift, m.sLength, context.blastProgram, BlastProgramTag<p>());
+
+    int8_t const  qStepOne = (m.qFrameShift < 0) ?  -1 : 1;
+    int8_t const  sStepOne = (m.sFrameShift < 0) ?  -1 : 1;
+    int8_t const     qStep = qIsTranslated(context.blastProgram, BlastProgramTag<p>()) ? qStepOne * 3 : qStepOne;
+    int8_t const     sStep = sIsTranslated(context.blastProgram, BlastProgramTag<p>()) ? sStepOne * 3 : sStepOne;
+
+    auto const & row0        = row(m.align, 0);
+    auto const & row1        = row(m.align, 1);
+
+    TPos const   maxPos      = std::max({effQStart, effQEnd, effSStart, effSEnd});
+    // max # digits in pos's
+    unsigned char const numberWidth = _numberOfDigits(maxPos);
+
+    while (aPos < m.alignStats.alignmentLength)
+    {
+        // Query line
+        sprintf(buffer, "Query  %-*d  ", numberWidth, qPos + effQStart);
```

Comment:
I have removed this in all places, except the one above, because this behaviour cannot be achieved with our `FormattedNumber`.